### PR TITLE
fix: parse arbitrum bridge transactions

### LINKED ISSUE #7200: Parse arbitrum bridge transactions properly
### Overview

We want to parse the abitrum bridge transactions properly so our user can easily identify which of these transactions are:
- Withdraw request from Arbitrum to Ethereum on both Native and ERC20 side
- Withdraw request from Ethereum to Arbitrum on both Native and ERC20 side

For now we don't really need to parse Claim transactions as we can't claim from our interface

Here is what we currently see:
image

And here is what is expected:
image

### References and additional details

For this you'll need to implement multiple parsers to `unchained-client`, one for the arbitrum side and another on the ethereum side.

### Acceptance Criteria

- All the transactions except claiming related with the arbitrum bridge are parsed and displayed in an human friendly way.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description

- I did try to respect all the naming the arbitrum team gave to their contracts so it's easier to understand which contracts we still use while parsing
- There is one TX ([ETH receive on Arbitrum side](https://arbiscan.io/tx/0x007a4576b494bee2bef429a5b53ac029a5c7414aee6ea9bcc9ccf1f3b17803fb)) that looks impossible to parse without further works: We might need to match back the `from` address with the initial TX to match that this is an arbitrumBridge tx
- Added every tests for the parsed txs on both ethereum and arbitrum side
- copies are a wip, we might need the @shapeshift/product to decide on those (see screenshots)

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #7200

## Risk
Low risk, presentational

## Testing
Using the arbitrum bridge:
- Do native token L1 to L2 deposit
- Do ERC20 token L1 to L2 deposit (maybe using 2 different tokens so we can verify that it works for most of the erc20)
- Do ETH L2 to L1 withdraw
- Do ERC20 L2 to L1 withdraw

### Engineering
n/a

### Operations
n/a

### Arbitrum side
image

### Ethereum side
image

### CODE REVIEW

**Review by gomesalexandre — COMMENTED** [review]
Tested locally, conceptually happy with this guy, see testing below.

Happy to get this guy in as-is, a few smallish issues to tackle at your convenience, with one preferably blocking issue and two suggestions for a follow-up

- preferably-blocking: We should standardize copies and make them more user-friendly i.e something like "Bridge to/from Arbitrum" vs. "Deposit/Withdraw" which may be confusing for users
- suggestion: we could surface the <amount> <token> deposited/withdrawn
- suggestion: not sure if that's something we're able to parse that, but https://www.rollup.codes/arbitrum-one may be the goer as far as parsing ETH deposit outbound Txs (cc @kaladinlight, does unchained expose that)

- ETH Deposit Request

https://etherscan.io/tx/0xccf99c2bbcae1a6ee80bf1b8a9d569ca550f1e44f87611ed66005e6b1f909989
image

- ERC20 Deposit Request

image

Feels weird to have ETH/FOX here (I'd expect to see "1 FOX", and same for all operations) but happy to have this as a follow-up improvement - as discussed here, the guts of this is to first get parsing happy, and then surface more data!

- ERC20 Deposit Received

image

- ERC20 Withdraw Request

image
https://arbiscan.io/tx/0xd6e8a9e9d6a0a0e15be54c12620603860db2086acbc8b4a4babc885eefaf6541

- ETH Withdraw Request

image
https://arbiscan.io/tx/0x35779b5c1d5e18534558dae6e59f3577337e5722224e6d66ac2915323cea4f56

File: packages/unchained-client/src/evm/parser/arbitrumBridge.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      !txInteractsWithContract(tx, ARB_RETRYABLE_TX_CONTRACT) &&
+      !txInteractsWithContract(tx, L2_ARBITRUM_CUSTOM_GATEWAY_CONTRACT) &&
+      !txInteractsWithContract(tx, ARBITRUM_L2_ERC20_GATEWAY_PROXY) &&
+      !txInteractsWithContract(tx, L1_ORBIT_CUSTOM_GATEWAY_CONTRACT)
+    )
+      return
+
+    if (!tx.inputData) return
+
+    let selectedAbi = (() => {
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply packages/unchained-client/src/evm/parser/arbitrumBridge.ts]: done in [08cc754](https://github.com/shapeshift/web/pull/7201/commits/08cc754f7dc2cf5bb6e4f506ca795e45e0cf582b)

File: packages/unchained-client/src/evm/parser/arbitrumBridge.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import type { Tx } from './types'
+
+export const ARB_SYS_CONTRACT = '0x0000000000000000000000000000000000000064'
+export const ARBITRUM_L2_ERC20_GATEWAY_PROXY = '0x09e9222E96E7B4AE2a407B98d48e330053351EEe'
+
+export const ARB_RETRYABLE_TX_CONTRACT = '0x000000000000000000000000000000000000006e'
+export const L2_ARBITRUM_CUSTOM_GATEWAY_CONTRACT = '0x096760F208390250649E3e8763348E783AEF5562'
+export const L2_ARBITRUM_GATEWAY_CONTRACT = '0x5288c571Fd7aD117beA99bF60FE0846C4E84F933'
+export const L1_ARBITRUM_GATEWAY_CONTRACT = '0x72ce9c846789fdb6fc1f34ac4ad25dd9ef7031ef'
+export const L1_ORBIT_CUSTOM_GATEWAY_CONTRACT = '0x4Dbd4fc535Ac27206064B68FfCf827b0A60BAB3f'
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply packages/unchained-client/src/evm/parser/arbitrumBridge.ts]: done in [08cc754](https://github.com/shapeshift/web/pull/7201/commits/08cc754f7dc2cf5bb6e4f506ca795e45e0cf582b)

File: packages/unchained-client/src/evm/ethereum/parser/__tests__/ethereum.test.ts [hunk 3]
```diff
@@ -3096,4 +3099,87 @@ describe('parseTx', () => {
...
+        transfers: [],
+        txid: '0xd115a9c89b5a387fc4da3be84329038e25cb13e36dd054126de9ac9ae3177f19',
+      }
+      const txParser = await makeTxParser()
+      const actual = await txParser.parse(tx, address)
+
+      expect(actual).toEqual(expected)
+    })
+
+    it('should be able to parse native create retryable ticket', async () => {
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply packages/unchained-client/src/evm/ethereum/parser/__tests__/ethereum.test.ts]: done in [08cc754](https://github.com/shapeshift/web/pull/7201/commits/08cc754f7dc2cf5bb6e4f506ca795e45e0cf582b)

File: packages/unchained-client/src/evm/ethereum/parser/__tests__/ethereum.test.ts [hunk 4]
```diff
@@ -3096,4 +3099,87 @@ describe('parseTx', () => {
...
+        transfers: [],
+        txid: '0x2b666f87986ed222e45c5299f893258e2468d59068ed36425f26ee2088baeb60',
+      }
+      const txParser = await makeTxParser()
+      const actual = await txParser.parse(tx, address)
+
+      expect(actual).toEqual(expected)
+    })
+
+    it('should be able to parse native deposit', async () => {
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply packages/unchained-client/src/evm/ethereum/parser/__tests__/ethereum.test.ts]: done in [08cc754](https://github.com/shapeshift/web/pull/7201/commits/08cc754f7dc2cf5bb6e4f506ca795e45e0cf582b)

File: packages/unchained-client/src/evm/arbitrum/parser/__tests__/arbitrum.test.ts [hunk 5]
```diff
@@ -1029,4 +1036,184 @@ describe('parseTx', () => {
...
+        status: 'Confirmed',
+        trade: undefined,
+        transfers: [],
+        txid: '0x7c22a8568b2703db2c87ad60ba1d6ef346012b0d9866ac6ae8e21a1b8b9089eb',
+      }
+      const actual = await txParser.parse(tx, address)
+      expect(actual).toEqual(expected)
+    })
+
+    it('should be able to parse erc20 l1 native withdraw request', async () => {
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply packages/unchained-client/src/evm/arbitrum/parser/__tests__/arbitrum.test.ts]: done in [08cc754](https://github.com/shapeshift/web/pull/7201/commits/08cc754f7dc2cf5bb6e4f506ca795e45e0cf582b)

File: packages/unchained-client/src/evm/arbitrum/parser/__tests__/arbitrum.test.ts [hunk 6]
```diff
@@ -1029,4 +1036,184 @@ describe('parseTx', () => {
...
+        trade: undefined,
+        transfers: [],
+        txid: '0x75f20f8f2caead042ac63c3175f34e505bc9a388a10b4c4e64a55831ef4d33cf',
+      }
+
+      const actual = await txParser.parse(tx, address)
+      expect(actual).toEqual(expected)
+    })
+
+    it('should be able to parse erc20 l1 native deposit', async () => {
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply packages/unchained-client/src/evm/arbitrum/parser/__tests__/arbitrum.test.ts]: Doesn't exist anymore in this file done in [08cc754](https://github.com/shapeshift/web/pull/7201/commits/08cc754f7dc2cf5bb6e4f506ca795e45e0cf582b)

File: packages/unchained-client/src/evm/arbitrum/parser/__tests__/arbitrum.test.ts [hunk 7]
```diff
@@ -1029,4 +1036,184 @@ describe('parseTx', () => {
...
+        trade: undefined,
+        transfers: [],
+        txid: '0xd115a9c89b5a387fc4da3be84329038e25cb13e36dd054126de9ac9ae3177f19',
+      }
+
+      const actual = await txParser.parse(tx, address)
+      expect(actual).toEqual(expected)
+    })
+
+    it('should be able to parse erc20 l1 native receive', async () => {
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply packages/unchained-client/src/evm/arbitrum/parser/__tests__/arbitrum.test.ts]: done in [08cc754](https://github.com/shapeshift/web/pull/7201/commits/08cc754f7dc2cf5bb6e4f506ca795e45e0cf582b)

File: packages/unchained-client/src/evm/arbitrum/parser/__tests__/arbitrum.test.ts [hunk 8]
```diff
@@ -1029,4 +1036,184 @@ describe('parseTx', () => {
...
+        confirmations: 330563,
+        data: {
+          assetId: undefined,
+          method: 'depositEth',
+          parser: 'arbitrumBridge',
+        },
+        status: 'Confirmed',
+        trade: undefined,
+        transfers: [],
+        txid: '0xd115a9c89b5a387fc4da3be84329038e25cb13e36dd054126de9ac9ae3177f19',
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply packages/unchained-client/src/evm/arbitrum/parser/__tests__/arbitrum.test.ts]: done in [08cc754](https://github.com/shapeshift/web/pull/7201/commits/08cc754f7dc2cf5bb6e4f506ca795e45e0cf582b)

File: packages/unchained-client/src/evm/arbitrum/parser/__tests__/arbitrum.test.ts [hunk 9]
```diff
@@ -1029,4 +1036,184 @@ describe('parseTx', () => {
...
+        confirmations: 15778,
+        data: {
+          assetId: undefined,
+          method: 'createRetryableTicket',
+          parser: 'arbitrumBridge',
+        },
+        status: 'Confirmed',
+        trade: undefined,
+        transfers: [],
+        txid: '0x75f20f8f2caead042ac63c3175f34e505bc9a388a10b4c4e64a55831ef4d33cf',
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply packages/unchained-client/src/evm/arbitrum/parser/__tests__/arbitrum.test.ts]: done in [08cc754](https://github.com/shapeshift/web/pull/7201/commits/08cc754f7dc2cf5bb6e4f506ca795e45e0cf582b)

**Comment by NeOMakinG** [comment]
> Tested locally, conceptually happy with this guy, see testing below.
> 
> Happy to get this guy in as-is, a few smallish issues to tackle at your convenience, with one preferably blocking issue and two suggestions for a follow-up
> 
> * preferably-blocking: We should standardize copies and make them more user-friendly i.e something like "Bridge to/from Arbitrum" vs. "Deposit/Withdraw" which may be confusing for users
> * suggestion: we could surface the   deposited/withdrawn
> * suggestion: not sure if that's something we're able to parse that, but https://www.rollup.codes/arbitrum-one may be the goer as far as parsing ETH deposit outbound Txs (cc @kaladinlight, does unchained expose that)
> * ETH Deposit Request
> 
> https://etherscan.io/tx/0xccf99c2bbcae1a6ee80bf1b8a9d569ca550f1e44f87611ed66005e6b1f909989 image
> 
> * ERC20 Deposit Request
> 
> image
> Feels weird to have ETH/FOX here (I'd expect to see "1 FOX", and same for all operations) but happy to have this as a follow-up improvement - as discussed here, the guts of this is to first get parsing happy, and then surface more data!
> 
> * ERC20 Deposit Received
> 
> image
> * ERC20 Withdraw Request
> 
> image https://arbiscan.io/tx/0xd6e8a9e9d6a0a0e15be54c12620603860db2086acbc8b4a4babc885eefaf6541
> * ETH Withdraw Request
> 
> image https://arbiscan.io/tx/0x35779b5c1d5e18534558dae6e59f3577337e5722224e6d66ac2915323cea4f56

Copies updated in [8b3857f](https://github.com/shapeshift/web/pull/7201/commits/8b3857ff80fc294b2d891d1540de1d6fc382dbc1)

For the rest, happy to tackle it in another PR if we want to

**Review by gomesalexandre — APPROVED** [review]
Comments have been tackled - retested locally, Tx parsing still LGTM and much better now with the updated copies.

Amazing job ser 🐐

**Pull request merged**